### PR TITLE
MSBuild fork of ResX reader/writer

### PR DESCRIPTION
**Pull request by rainersigwald**
Related: #308, #2221, https://github.com/dotnet/corefx/issues/34204.

ResX types are available and functional on .NET Core, but are in the winforms layer, which we don't want MSBuild to have to be on top of. Fortunately, they're pretty isolated, so we can pull a fork of that source into our repo and use it.

This will also probably help with future work to avoid deserialization/serialization during resource building, but doesn't directly address that yet.

Recommended review strategy:
* Commit-by-commit
* Ignore 7d32eb4 because it was just copying winforms stuff

To do:

- [x] Pull in source
- [x] make it compile
- [x] update to latest source
- [x] ResXFileRef
- [ ] Validate incremental behavior of same

### CODE REVIEW

**Comment by rainersigwald** [comment]
Marking as WIP since https://github.com/dotnet/winforms/pull/212 just landed and I should adopt it too.

File: src/Tasks/winforms/misc/MultitargetUtil.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                if (string.IsNullOrEmpty(assemblyQualifiedName))
+                {
+                    assemblyQualifiedName = type.AssemblyQualifiedName;
+                }
+            }
+
+            return assemblyQualifiedName;
+        }
+
+        // ExecutionEngineException is obsolete and shouldn't be used (to catch, throw or reference) anymore.
```
Comment (danmosemsft) [inline src/Tasks/winforms/misc/MultitargetUtil.cs]: You can just remove ExecutionEngineException here.

**Comment by danmosemsft** [comment]
If you didn't already, you might checkout https://github.com/dotnet/corefx/compare/master...danmosemsft:resx2?expand=1 and diff those files against yours, as I know I made various cleanup changes, possibly something will pop out that should come along. Or not.

**Comment by rainersigwald** [comment]
FYI @merriemcgaw, @dreddy-work

**Comment by rainersigwald** [comment]
This is no longer the approach we're going with. I'll put up a new PR shortly with an implementation that uses https://github.com/dotnet/corefx/pull/36906.

**Pull request closed without merging**